frodo.js

```diff
@@ -1,75 +1,81 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.